core/src/main/scala/cromwell/core/logging/WorkflowLogger.scala

```diff
@@ -22,12 +21,29 @@ trait WorkflowLogging extends ActorLogging { this: Actor =>
 }
 
 object WorkflowLogger {
-  def makeFileLogger(path: Path, level: Level): Logger = synchronized { // see below
+  def makeFileLogger(path: Path, level: Level): Logger = {
+    // see below
     val ctx = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
     val name = path.getFileName.toString
 
     Option(ctx.exists(name)) match {
       case Some(existingLogger) => existingLogger
+      case None => makeSynchronizedFileLogger(path, level, ctx, name)
+    }
+  }
+
+  /*
+  logback-core 1.1.6 started iterating over an internal hashmap, and now throws a ConcurrentModificationException
+  if two appenders call start() at the same time.
+
+  So now we synchronize makeFileLogger(). Could also try just synchronizing just the appender.start().
+
+  https://github.com/qos-ch/logback/commit/77128a003a7fd7e8bd7a6ddb12da7a65cf296593#diff-f8cd32379a53986c2e70e2abe86fa0faR145
+   */
+  private def makeSynchronizedFileLogger(path: Path, level: Level, ctx: LoggerContext, name: String): Logger =
+  synchronized {
+    Option(ctx.exists(name)) match {
```

Comment:
While the inner map is the one that throws a concurrent modification exception, I wanted to avoid creating two loggers for the same name, like we used to. So to avoid that case this code now:

- checks the logger doesn't already exist
- waits for the mutex
- re-checks that the logger doesn't already exist
- creates the logger if needed
- releases the mutex